kong/core/cluster.lua

```diff
@@ -25,29 +40,23 @@ local function async_autojoin(premature)
   -- If this node is the only node in the cluster, but other nodes are present, then try to join them
   -- This usually happens when two nodes are started very fast, and the first node didn't write his
   -- information into the datastore yet. When the second node starts up, there is nothing to join yet.
-  local lock, err = resty_lock:new("cluster_autojoin_locks", {
-    exptime = ASYNC_AUTOJOIN_INTERVAL - 0.001
-  })
-  if not lock then
-    ngx_log(ngx.ERR, "could not create lock: ", err)
-    return
-  end
-  local elapsed = lock:lock("async_autojoin")
-  if elapsed and elapsed == 0 then
+  local ok, err = get_lock(ASYNC_AUTOJOIN_KEY, ASYNC_AUTOJOIN_INTERVAL)
+  if ok then
+    log_debug("auto-joining")
```

Comment:
ditto: maybe missing some context for a regular user reading this (even with the log prefix)